docs/details/signal.dox

```diff
@@ -211,32 +211,82 @@ respectively, then the possible batch operations are as follows.
 \defgroup signal_func_approx1 approx1
 \ingroup approx_mat
 
-approx1 interpolates data along the first dimensions.
-It has three options for the type of interpolation to perform:
-- Nearest neighbor  - \ref AF_INTERP_NEAREST
-- Linear interpolation  - \ref AF_INTERP_LINEAR
-- Bilinear interpolation - \ref AF_INTERP_BILINEAR
-- Cubic interpolation - \ref AF_INTERP_CUBIC
+approx1 performs interpolation on data along one dimension.
 
-Interpolation is performed assuming input data is equally spaced with indices
-in the range [0, n). The positions are sampled with respect to data at these
-locations.
+The input array, `in`, is comprised of values which lie on a range of
+indices defined by `idx_start` and `idx_step` [idx_start:idx_step:N]
+along dimension `interp_dim`.
+
+The positions array, `pos`, contains indices whose values we want to
+find, given the input array. Values of **known indices** will be
+looked up in the input array, while values of **unknown indices** will
+be found via interpolation. Indices outside of the index range are not
+extrapolated. Instead, those values are set to `off_grid`, whose
+default value is 0.
+
+Several types of interpolation are supported by `approx1`:
+
+- Nearest neighbor interpolation - \ref AF_INTERP_NEAREST
+- Linear interpolation (default) - \ref AF_INTERP_LINEAR, \ref AF_INTERP_LINEAR_COSINE
+- Cubic interpolation - \ref AF_INTERP_CUBIC, \ref AF_INTERP_CUBIC_SPLINE
+- Lower interpolation - \ref AF_INTERP_LOWER
+
+Unless specified, linear interpolation is performed by default. Refer
+to \ref af_interp_type for more information about ArrayFire's
+interpolation types.
 
 \snippet test/approx1.cpp ex_signal_approx1
 
 \defgroup signal_func_approx2 approx2
 \ingroup approx_mat
 
-approx2 performs interpolation on data along the first and second dimensions.
-It has three options for the type of interpolation to perform:
-- Nearest neighbor  - \ref AF_INTERP_NEAREST
-- Linear interpolation  - \ref AF_INTERP_LINEAR
-- Bilinear interpolation - \ref AF_INTERP_BILINEAR
-- Cubic interpolation - \ref AF_INTERP_CUBIC
-
-Interpolation is performed assuming input data is equally spaced with indices
-in the range [0, n) along each dimension.
-The positions are sampled with respect to data at these locations.
+approx2 interpolates data along two dimensions. Interpolation computes
+for values of unknown points out of known ones.
```

Comment:
The second sentence is not needed, this makes the short description too long to appear in UI - check the very first sentence in [this](http://arrayfire.org/docs/group__signal__func__approx1.htm) page. Also, you talk about what interpolation does below also. So, this is not needed here in my opinion.